pkg/gadgets/run/tracer/tracer.go

```diff
@@ -177,6 +179,78 @@ func (t *Tracer) Close() {
 	}
 }
 
+var (
+	once             sync.Once
+	ringbufAvailable bool
+)
+
+func isRingbufAvailable() bool {
+	once.Do(func() {
+		_, err := ebpf.NewMap(&ebpf.MapSpec{
+			Type:       ebpf.RingBuf,
+			MaxEntries: uint32(os.Getpagesize()),
+		})
+
+		ringbufAvailable = err == nil
+	})
+
+	return ringbufAvailable
+}
+
+func (t *Tracer) loadeBPFObjects(opts ebpf.CollectionOptions, tracerMapName string) error {
+	available := isRingbufAvailable()
+
+	if !available {
+		log.Debugf("Ring buffers are not available, defaulting to perf ones")
+		if tracerMapName != "" {
+			bufMap, ok := t.spec.Maps[tracerMapName]
+			if !ok {
+				return fmt.Errorf("no buffer map named %s", tracerMapName)
+			}
+
+			bufMap.Type = ebpf.PerfEventArray
+			bufMap.KeySize = 4
+			bufMap.ValueSize = 4
+
+			t.spec.Maps[tracerMapName] = bufMap
+		}
+	} else {
+		heapMapName := "gadget_heap"
+		// If we delete the gadget_heap map, the verifier will not load the code
+		// because it cannot find it despite the code using it being dead one:
+		// ...: create BPF collection: program ig_mount_x: missing map gadget_heap
+		// Rather than deleting it, we just set its size to 0 and use a hash to
+		// avoid having one struct per CPU, i.e. reducing as much as possible the
+		// memory footprint it uses.
+		heapMap, ok := t.spec.Maps[heapMapName]
+		if !ok {
+			return fmt.Errorf("no heap map named %s", heapMapName)
```

Comment:
I don't think it has to be an error, if the map doesn't exist, just ignore it and keep going.